splitio/sync/synchronizer.py

```diff
@@ -233,7 +233,13 @@ def synchronize_splits(self, till):
         """
         _LOGGER.debug('Starting splits synchronization')
         try:
-            self._split_synchronizers.split_sync.synchronize_splits(till)
+            segment_list = self._split_synchronizers.split_sync.synchronize_splits(till, self._split_synchronizers.segment_sync)
+            if len(segment_list) != 0:
```

Comment:
i think that if the list is not empty we should fetch the new segemnts in bg. we should spawn a thread, set it to daemon mode, and then fetch the segments sequentially inside it

Reply:
We are adding the new segments to the workerpool so the syncing will happen in bg, any advantage adding new code to perform the bg synching?

Reply:
Looks like we are good with using workerpool and will only wait for completion when called from sync_now(), otherwise the workerpool will sync the segment(s) in the background, and the sync function return True